xbmc/interfaces/legacy/ModuleXbmc.cpp

```diff
@@ -322,6 +333,43 @@ namespace XBMCAddon
       return infoMgr.GetImage(ret, WINDOW_INVALID);
     }
 
+    String getDatabaseName(const char * dbtype)
+    {
+      XBMC_TRACE;
+      String result;
+
+      if (StringUtils::CompareNoCase(dbtype, "addon") == 0)
+      {
+        result = CServiceBroker::GetDatabaseManager().GetFullDatabaseName(CAddonDatabase::GetDefaultBaseDBName());
+      }
+      else if (StringUtils::CompareNoCase(dbtype, "epg") == 0)
+      {
+        result = CServiceBroker::GetDatabaseManager().GetFullDatabaseName(CPVREpgDatabase::GetDefaultBaseDBName());
+      }
+      else if (StringUtils::CompareNoCase(dbtype, "music") == 0)
+      {
+        result = CServiceBroker::GetDatabaseManager().GetFullDatabaseName(CMusicDatabase::GetDefaultBaseDBName());
+      }
+      else if (StringUtils::CompareNoCase(dbtype, "pvr") == 0)
+      {
+        result = CServiceBroker::GetDatabaseManager().GetFullDatabaseName(CPVRDatabase::GetDefaultBaseDBName());
+      }
+      else if (StringUtils::CompareNoCase(dbtype, "texture") == 0)
+      {
+        result = CServiceBroker::GetDatabaseManager().GetFullDatabaseName(CTextureDatabase::GetDefaultBaseDBName());
+      }
+      else if (StringUtils::CompareNoCase(dbtype, "video") == 0)
+      {
+        result = CServiceBroker::GetDatabaseManager().GetFullDatabaseName(CVideoDatabase::GetDefaultBaseDBName());
+      }
+      else if (StringUtils::CompareNoCase(dbtype, "viewmode") == 0)
+      {
+        result = CServiceBroker::GetDatabaseManager().GetFullDatabaseName(CViewDatabase::GetDefaultBaseDBName());
+      }
```

Comment:
This code looks (despite from case insensitive compare which should be fixed) exactly like the json-rpc impl. I think we should write a helper function which can be called from both python and json-rpc impl.

Home for this function could be a new static member function in `utils/DatabaseUtils.h`.